packages/core/src/components/forms/_file-input.scss

```diff
@@ -143,6 +143,11 @@ $file-input-button-padding-large: ($pt-input-height-large - $pt-button-height) /
   }
 }
 
+#blueprint-documentation [data-example-id="file-input"] .#{$ns}-file-upload-input::after {
+  content: "Browse";
```

Comment:
why do you need this? shouldn't it be set as the default in the JS component?

Reply:
I still don't think this block is needed, it doesn't seem to do anything

Reply:
I added this to this block. Attr function has the fallback parameter, but it's not supported by browsers. How I can set style only for docs?
<img width="795" alt="Screenshot 2019-06-18 at 23 45 07" src="https://user-images.githubusercontent.com/27079794/59718627-39a2ec00-9223-11e9-8198-9e9eb6a4a4ae.png">

Reply:
`attr()` is supported for the `content` property by all the browsers we care about: https://caniuse.com/#feat=css-gencontent

not sure what you're trying to achieve here? this kind of selector _would_ be the right way to do docs-only styles, but there are a couple issues

- the styles should go in docs-app along with the other overrides; not in blueprint core
- the `data-example-id` selector is wrong

and I still think you don't need it overall

Reply:
I meant the second argument of the function attr (fallback)
If you don't use this style, the text is not displayed:
<img width="507" alt="Screenshot 2019-06-19 at 00 13 30" src="https://user-images.githubusercontent.com/27079794/59720368-5b9e6d80-9227-11e9-8153-5ed48421d49b.png">
I moved rule in docs styles

Reply:
hmm, that's not good then... we don't want all users to have to add a style override just to get things working out of the box... and unfortunately I don't want to introduce a breaking change to the CSS API where it's _required_ to supply a `bp3-button-text` attribute in order to get the text to render...

can we try using multiple CSS statements to achieve the fallback? if not, we might have to delay merging this PR as it could be a breaking change...

```css
content: "Browse";
content: attr(#{$ns}-button-text);
```

Reply:
Maybe I wrote something wrong. Sorry for my English.
This rule is only needed for the CSS block in the documentation. In other places it isn't needed and everything works well.

Reply:
If you write like this, the second rule will overwrite the first. And there will be an empty string
```css
content: "Browse";
content: attr (#{$ns}-button-text);
```

Reply:
Your English is fine, no need to apologize :)

> This rule is only needed for the CSS block in the documentation.

Unfortunately, this is not true for this library because the CSS-only styling API is part of Blueprint's public API (in addition to the JS components).

Some users render this markup to get a working file input with the word "Browse":

```html
<label class="bp3-file-input">
  <input type="file"/>
  <span class="bp3-file-upload-input">Choose file...</span>
</label>
```

We don't want to remove support for this, otherwise it would be a regression... even though I understand that it's a minor syntax migration to change the API to this instead:

```html
<label class="bp3-file-input">
  <input type="file"/>
  <span class="bp3-file-upload-input" bp3-button-text="Browse">Choose file...</span>
</label>
```

... it is still a _breaking change_.

Reply:
I solved the problem. The CSS is just the string `content: "Borwse"`. When creating a component, a new style is created.
But for the documentation I had to leave the style, because the new style is applied.